src/plots/plots.js

```diff
@@ -809,6 +809,16 @@ plots.purge = function(gd) {
     // remove modebar
     if(fullLayout._modeBar) fullLayout._modeBar.destroy();
 
+    if(fullLayout._plots) {
```

Comment:
> so by the time cleanPlot is called, there's no longer information about what to clean.

Wow. That's bad. Nice catch.

I think the _best_ solution at the moment would be to add 

```js
 // remove gl contexts
Plots.cleanPlot([], {}, fullData, fullLayout);
```

to `Plotly.newPlot` before it calls `Plots.purge`.